app/src/main/java/com/tananaev/passportreader/MainActivity.java

```diff
@@ -426,7 +428,17 @@ private void doPassiveAuth() {
                     CertPathValidator cpv = CertPathValidator.getInstance(CertPathValidator.getDefaultType());
                     cpv.validate(cp, pkixParameters);
 
-                    Signature sign = Signature.getInstance(sodFile.getDigestEncryptionAlgorithm());
+                    String sodDigestEncryptionAlgorithm = sodFile.getDigestEncryptionAlgorithm();
+
+                    if (sodDigestEncryptionAlgorithm.equals("SSAwithRSA/PSS")) {
+                        sodDigestEncryptionAlgorithm = "SHA256withRSA/PSS";
+                    }
+
+                    Signature sign = Signature.getInstance(sodDigestEncryptionAlgorithm);
+                    if (sodDigestEncryptionAlgorithm.equals("SHA256withRSA/PSS")) {
```

Comment:
I don't really understand the logic here. Why can't you just have a single `if`?

Reply:
One `if` check would also work, then it would include the entire block that does the signature verification. I used two `if` checks as I thought it be more granular.

Reply:
But you only do signature check for this `if` case, unless I'm missing something.

Reply:
Signature check is also done in other cases, aside from this one. The app reported a `Passive Authentication` check of a French passport from a fellow developer, but failed in mine. Hence, I suspect there is more going on here in terms of signature algorithm types. That is, that French passport reported a different type (I have no debug, he just showed it via chat). Hence, the granularity of my two `if` checks.

Reply:
OK, in that case can we at least save `sodDigestEncryptionAlgorithm.equals("SSAwithRSA/PSS")` into a variable, so we don't need to do two string comparisons?

Reply:
Correct. A `boolean` flag would also do.